client/homescreen/activity-panel/reviews/index.js

```diff
@@ -0,0 +1,432 @@
+import { getAdminLink } from '@woocommerce/wc-admin-settings';
+import { get, isNull } from 'lodash';
+import { REVIEWS_STORE_NAME } from '@woocommerce/data';
+import { recordEvent } from '@woocommerce/tracks';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import {
+	ActivityCard,
+	ActivityCardPlaceholder,
+} from '../../../header/activity-panel/activity-card';
+import { CurrencyContext } from '../../../lib/currency-context';
+import sanitizeHTML from '../../../lib/sanitize-html';
+import { REVIEW_PAGE_LIMIT } from './utils';
+
+class ReviewsPanel extends Component {
+	recordReviewEvent( eventName, eventData ) {
+		recordEvent( `reviews_${ eventName }`, eventData || {} );
+	}
+
+	deleteReview( reviewId ) {
+		const {
+			deleteReview,
+			createNotice,
+			updateReview,
+			clearReviewsCache,
+			reviews,
+		} = this.props;
+		if ( reviewId ) {
+			deleteReview( reviewId )
+				.then( () => {
+					clearReviewsCache();
+					createNotice(
+						'success',
+						__(
+							'Review successfully deleted.',
+							'woocommerce-admin'
+						),
+						{
+							actions: [
+								{
+									label: __( 'Undo', 'woocommerce-admin' ),
+									onClick: () => {
+										updateReview(
+											reviewId,
+											{
+												deleted: false,
+											},
+											{
+												_embed: 1,
+											}
+										).then( () =>
+											clearReviewsCache(
+												reviews.length < 2
+											)
+										);
+									},
+								},
+							],
+						}
+					);
+				} )
+				.catch( () => {
+					createNotice(
+						'error',
+						__(
+							'Review could not be deleted.',
+							'woocommerce-admin'
+						)
+					);
+				} );
+		}
+	}
+
+	updateReviewStatus( reviewId, newStatus, oldStatus ) {
+		const {
+			createNotice,
+			updateReview,
+			clearReviewsCache,
+			reviews,
+		} = this.props;
+		if ( reviewId ) {
+			updateReview( reviewId, { status: newStatus } )
+				.then( () => {
+					clearReviewsCache();
+					createNotice(
+						'success',
+						__(
+							'Review successfully updated.',
+							'woocommerce-admin'
+						),
+						{
+							actions: [
+								{
+									label: __( 'Undo', 'woocommerce-admin' ),
+									onClick: () => {
+										updateReview(
+											reviewId,
+											{
+												status: oldStatus,
+											},
+											{
+												_embed: 1,
+											}
+										).then( () =>
+											clearReviewsCache(
+												reviews.length < 2
+											)
+										);
+									},
+								},
+							],
+						}
+					);
+				} )
+				.catch( () => {
+					createNotice(
+						'error',
+						__(
+							'Review could not be updated.',
+							'woocommerce-admin'
+						)
+					);
+				} );
+		}
+	}
+
+	renderReview( review ) {
+		const product =
+			( review &&
+				review._embedded &&
+				review._embedded.up &&
+				review._embedded.up[ 0 ] ) ||
+			null;
+
+		if ( review.isUpdating || this.props.isRequesting ) {
+			return (
```

Comment:
Showing loading placeholder when an review is updating or when it is updating the entire list.